Code/Framework/AzToolsFramework/AzToolsFramework/Entity/PrefabEditorEntityOwnershipService.cpp

```diff
@@ -443,6 +459,7 @@ namespace AzToolsFramework
 
     void PrefabEditorEntityOwnershipService::OnEntityRemoved(AZ::EntityId entityId)
     {
+        AZ_Assert(m_entitiesRemovedCallback, "Callback function for entity removal has not been set.");
```

Comment:
Is the callback always meant to be set? If yes then an assert is correct, if it's a choice to set it or not, then it seems better to remove the assert and do this instead:

````
    AzFramework::SliceEntityRequestBus::MultiHandler::BusDisconnect(entityId);
    if (m_entitiesRemovedCallback)
    {
        m_entitiesRemovedCallback({ entityId });
    }
````

Reply:
Yeah, the callback is always meant to be set.

Tested opening/closing level in editor and launcher.

Reply:
Thanks @michabr